vsphere/resource_vsphere_virtual_disk.go

```diff
@@ -173,16 +174,27 @@ func resourceVSphereVirtualDiskCreate(d *schema.ResourceData, meta interface{})
 		if directoryPathIndex > 0 {
 			vmdkPath := vDisk.vmdkPath[0:directoryPathIndex]
```

Comment:
A predefined sleep timer will probably work in most cases but is not sufficiently robust IMO.
You can declare a mutex (called `m` in the example below) and lock the time-sensitive code

```suggestion
			m.Lock()
			vmdkPath := vDisk.vmdkPath[0:directoryPathIndex]
			log.Printf("[DEBUG] Creating parent directories: %v", ds.Path(vmdkPath))
			err = fm.MakeDirectory(context.TODO(), ds.Path(vmdkPath), dc, true)
			...
			m.Unlock()
```

Terraform invokes the provider over RPC and all disk resources effectively share the same runtime during a `terraform apply`

Reply:
If I'm understanding what you are suggesting correctly, the mutex would basically ensure that only one `MakeDirectory` operation is allowed at a time, irrespective of the directory being created. I just wanted to check there isn't any concern about performance in the case multiple different directories need to be created (across multiple disks) where currently those would be created in parallel, but changing to using the mutex would result in them being created in series. I expect in the overall scheme of the time it takes to create a VM, the additional time resulting from creating them in series would be negligible (so using a mutex for its other advantages would be beneficial), but just wanted to get a second opinion. Thanks.